src/App.tsx

```diff
@@ -1,204 +1,58 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import React, { useEffect, useState } from 'react';
+import { getTodos } from './api/todos';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Todo } from './types/Todo';
+import { ErrorMessages } from './types/ErrorMessages';
+import { FilterOptions } from './types/FilterOptions';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterOption, setFilterOption] = useState<FilterOptions>(
+    FilterOptions.DEFAULT,
+  );
+  const [errorMessage, setErrorMessage] = useState<ErrorMessages>(
+    ErrorMessages.NO_ERROR,
+  );
+
+  const handleErrorReset = () => {
+    setErrorMessage(ErrorMessages.NO_ERROR);
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorMessages.UNABLE_TO_LOAD_TODO);
+        setTimeout(handleErrorReset, 3000);
+      });
+  }, []);
 
   return (
     <div className="todoapp">
       <h1 className="todoapp__title">todos</h1>
 
       <div className="todoapp__content">
-        <header className="todoapp__header">
-          {/* this button should have `active` class only if all todos are completed */}
-          <button
-            type="button"
-            className="todoapp__toggle-all active"
-            data-cy="ToggleAllButton"
+        <Header todos={todos} />
+        <TodoList todos={todos} filterOption={filterOption} />
+
+        {todos.length && (
```

Comment:
```suggestion
        {todos.length > 0 && (
```
If the todo length is 0, you will see 0 in the UI in your case